src/http/http.ml

```diff
@@ -668,7 +668,10 @@ let serve_with_maybe_https
 
 let default_interface = "localhost"
 let default_port = 8080
-let never = fst (Lwt.wait ())
+let on_sigterm =
+  let promise, resolve = Lwt.wait () in
+  ignore (Lwt_unix.on_signal Sys.sigterm (fun _ -> Lwt.wakeup_later resolve ()));
+  promise
```

Comment:
Creating this promise at program init time seems questionable now.

Before, the `never` promise was a sort of orphan promise in the heap that was fine to have floating around since init time.

But the `on_sigterm` promise should probably only be created if `serve`/`run` are actually called, since a program might end up with Dream linked in but never actually use it. It seems questionable to affect signal handling at init time even if Dream does not end up getting used (or at init time even if Dream does end up getting used).